### PR TITLE
make "toto = $(dir .)" better works on Windows

### PR DESCRIPTION
**Pull request by stonebig**

### CODE REVIEW

**Comment by stonebig** [comment]
it's a little bit shamefull that AppVeyor is not a "real user" test environnement

File: xonsh/tools.py [hunk 1]
```diff
@@ -37,8 +37,10 @@
     string_types = (str, unicode)
     unicode_type = unicode
 
-DEFAULT_ENCODING = sys.getdefaultencoding()
-
+if os.name == 'nt':
```
Comment (scopatz) [inline xonsh/tools.py]: Can you move this a couple of lines down and use the `ON_WINDOWS` variable, please?

**Comment by scopatz** [comment]
Thanks for putting this in @stonebig. Could you please document this change in behaviour in `docs/envvars.rst` too please?

**Comment by stonebig** [comment]
on a side remark, the dark blue color of the prompt is nearly unvisible in my dark black Windows terminal.

File: xonsh/tools.py [hunk 2]
```diff
@@ -46,6 +44,11 @@
 ON_POSIX = (os.name == 'posix')
 IS_ROOT = ctypes.windll.shell32.IsUserAnAdmin() != 0 if ON_WINDOWS else os.getuid() == 0
 
+if ON_WINDOWS == True and sys.stdout.encoding is not None:
```
Comment (scopatz) [inline xonsh/tools.py]: No need to have the `== True`.

**Comment by scopatz** [comment]
Could you also update the docs please? And yeah, the color issues should be fixed

**Comment by scopatz** [comment]
But don't worry about the colors here

**Comment by stonebig** [comment]
which docs ?

**Comment by scopatz** [comment]
`docs/envvars.rst.`

**Comment by scopatz** [comment]
Hey @stonebig, is this ready for another look?  Please ping the conversation whenever you think a PR is finished, so that we get emails about it.

**Comment by gforsyth** [comment]
Hey @stonebig -- is this ready for another look?

**Comment by stonebig** [comment]
hi @gforsyth.

I gave up for now until Jupyter 5 shows up with "promised" Prompt-toolkit integration. At that moment for sure, I'll give it another look.

**Comment by gforsyth** [comment]
Hey @stonebig -- IPython 5 has arrived with prompt_toolkit support, should we give this another crack?  I'm guessing it'll need an update from master :grinning:

**Comment by gforsyth** [comment]
OK, I'm closing this as it's a bit stale, but I would happily reopen if someone wants to take a crack at getting it up to date.

**Pull request closed without merging**